src/pykiso/lib/connectors/cc_serial.py

```diff
@@ -119,7 +125,60 @@ def __init__(
         )
 
         self.current_write_timeout = write_timeout
-        self.serial.port = port
+        self.serial.port = self._get_port(port=port, vid=vid, pid=pid, timeout=0.1)
+
+    def _find_device(vid: int, pid: int, timeout: float = 0.1) -> str:
+        """Return the device which matches pid and vid.
+
+        :param vid: vendor id
+        :param pid: product id
+        :param timeout: timeout in seconds to wait for at least one device
+            to be found.
+        :return: com port for the found device. I.e. "COM4" or "/dev/tty1"
+        """
+        start_time = time.perf_counter()
+
+        while time.perf_counter() - start_time < timeout:
+            attached_com_devices = serial.tools.list_ports.comports()
+            found_devices = [
+                port
+                for port in attached_com_devices
+                if port.pid == pid and port.vid == vid
+            ]
+            if found_devices:
+                break
+            time.sleep(0.1)
+        else:
+            raise TimeoutError(
+                f"Failed to detect connected USB device with IDs {vid:04X}:{pid:04x} within {timeout}s"
+            )
+
+        found_devices = [
+            port.device if sys.platform.startswith("linux") else port.name
+            for port in found_devices
+        ]
+        if len(found_devices) > 1:
+            log.warning(
+                f"Found multiple devices with matching IDs {vid:04X}:{pid:04x}. Select first device {found_devices[0]}."
```

Comment:
Maybe print info at which com ports those devices are attached to

Reply:
done